sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -374,6 +374,7 @@ case class AlterTableChangeColumnCommand(
   // TODO: support change column name/dataType/metadata/position.
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
+    catalog.refreshTable(tableName)
```

Comment:
let's add some code comments to explain the reason: this command may change column default values and we need to refresh the table relation cache so that DML commands can resolve default values correctly.

Reply:
This is done.

Reply:
Shall we call catalog.refreshTable(tableName) after check the default values changed?

Reply:
Hi @beliefer thanks for your comment, I think we have to check it before calling `catalog.getTableRawMetadata` so that method performs a catalog lookup as expected rather than reading from the cache (which may be stale). There's only one call to `catalog.getTableRawMetadata` in this function so we should be OK here, but thanks for keeping this question in mind :)